tiledb/bioimg/visualization/helpers.py

```diff
@@ -0,0 +1,65 @@
+import json
+from dataclasses import dataclass
+from typing import List, Optional, Sequence
+
+import tiledb
+
+from ..converters.axes import Axes
+
+
+@dataclass(frozen=True)
+class ZoomLevelRecord:
+    zoom_level: int
+    image_level: int
+    downsample: int
+    width: int
+    height: int
+    namespace: str
+    array_uri: str
+    axes: str
+
+
+def calculate_metadata(
+    namespace: str, group_uri: str, config: Optional[tiledb.Config] = None
+) -> Sequence[ZoomLevelRecord]:
+    """
+    Calculates the required metadata for the Tile Image viewer.
+
+    :param namespace: the namespace of the group
+    :param group_uri: path to the group
+    :param config: the configuration for constructing the tileDB context
+    """
+    metadata: List[ZoomLevelRecord] = []
+
+    context = None if config is None else tiledb.Ctx(config)
+    group = tiledb.Group(group_uri, "r", ctx=context)
+    zoom_level = 0
+
+    for meta in json.loads(group.meta["levels"])[::-1]:
+        axes = Axes(meta["axes"])
+        axes_mapper = axes.mapper(Axes("XYC"))
+        shape = axes_mapper.map_shape(meta["shape"])
+
+        zoom_factor = round(shape[0] / metadata[-1].width) if len(metadata) > 0 else 2
```

Comment:
1. A `width, height, channel = shape` unpacking would make things a bit clearer.
2. `if len(metadata) > 0` is more idiomatically spelled as `if metadata`